src/sqlancer/postgres/ast/PostgresJoin.java

```diff
@@ -15,18 +17,36 @@ public static PostgresJoinType getRandom() {
 
     }
 
-    private final PostgresTable table;
+    private final PostgresTableReference tableReference;
     private final PostgresExpression onClause;
     private final PostgresJoinType type;
 
-    public PostgresJoin(PostgresTable table, PostgresExpression onClause, PostgresJoinType type) {
-        this.table = table;
+    public static class PostgresTableReference implements PostgresExpression {
```

Comment:
I'm sorry, but I actually overlooked that a `PostgresFromTable` class already existed. This is what I had in mind when proposing the `PostgresTableReference` class, since it basically allows to treat a subquery and table uniformly as an expression. Would it be possible to remove this class? Sorry again.